_source/_posts/2022-01-26-webassembly-on-kubernetes-with-rust.md

```diff
@@ -0,0 +1,365 @@
+---
+layout: blog_post
+title: "Containerless! How to Run WebAssembly Workloads on Kubernetes with Rust"
+author: deepu-sasidharan
+by: advocate
+communities: [devops, rust]
+description: "Learn how to run secure WebAssembly workloads written with Rust on Kubernetes with Krustlet."
+tags: [rust, webassembly, kubernetes, security]
+tweets:
+  - "Do you want to go Containerless on Kubernetes? Learn how to run WebAssembly workloads written in Rust on Kubernetes. #rust #webassembly #kubernetes #security"
+  - "Learn how to run secure WebAssembly workloads written with Rust on Kubernetes with Krustlet. #rust #webassembly #kubernetes #security"
+  - "Is WebAssembly the future of Kubernetes? Learn all about Wasm on Kubernetes with Rust. #rust #webassembly #kubernetes #security"
+image: blog/webassembly-on-kubernetes-with-rust/rust_wasm_k8s.jpg
+type: awareness
+---
+
+WebAssembly (Wasm) is one of the most exciting and underestimated software technologies invented in recent times. It's a binary instruction format for a stack-based virtual machine. It aims to execute at native speeds with a memory-safe and secure sandbox. Its portable, cross-platform, and language-agnostic. It's designed as a compilation target for languages. Though originally part of the open web platform, it has found use cases beyond the web. WebAssembly is now used in browsers, Node.js, Deno, Kubernetes, and IoT platforms.
+
+You can learn more about WebAssembly at [WebAssembly.org](https://webassembly.org/).
+
+# WebAssembly on Kubernetes
+
+WebAssembly, though initially designed for [the web](https://www.w3.org/wasm/), proved to be an ideal format for writing platform and language agnostic applications. You are used to something similar in the container world, Docker containers. People figured WebAssembly is so similar and even more efficient since it's fast, portable, and secure, running at native speeds. This means that you can use WebAssembly alongside containers as workloads on Kubernetes. This was made possible by another WebAssembly initiative known as [WebAssembly System Interface (WASI)](https://wasi.dev/) and the [Wasmtime](https://github.com/bytecodealliance/wasmtime) project.
+
+{% twitter 1111004913222324225 %}
+
+WebAssembly on Kubernetes is relatively new and has some rough edges at the moment, but it's already proving to be revolutionary. Wasm workloads can be extremely fast as they can execute faster than a container takes to start; they are sandboxed and hence much more secure than containers; they are way smaller in size due to the binary format than containers.
+
+If you want to learn more about WASI, check out [the announcement from Mozilla](https://hacks.mozilla.org/2019/03/standardizing-wasi-a-webassembly-system-interface/).
```

Comment:
```suggestion
If you want to learn more about WASI, check out [the original announcement from Mozilla](https://hacks.mozilla.org/2019/03/standardizing-wasi-a-webassembly-system-interface/).
```
clarification since the announcement is a few years old now...